packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -405,6 +405,41 @@ export const Table = <
     return flatColumns.some((column) => column.id === SELECTION_CELL_ID);
   }, [columns]);
 
+  const disableUserSelect = React.useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === 'Shift') {
+        ownerDocument &&
+          (ownerDocument.documentElement.style.userSelect = 'none');
+      }
+    },
+    [ownerDocument],
+  );
+
+  const enableUserSelect = React.useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === 'Shift') {
+        ownerDocument && (ownerDocument.documentElement.style.userSelect = '');
+      }
+    },
+    [ownerDocument],
+  );
+
+  React.useEffect(() => {
+    // Add onkeydown event listeners (during creation)
+    if (selectionMode === 'multi') {
+      document.addEventListener('keydown', disableUserSelect);
```

Comment:
You should use `ownerDocument` here

Reply:
**[DONE]** Swapped `document` with `ownerDocument`, thanks. But just curious, what is the difference between the two? Also, just to note, `ownerDocument` can be undefined but I don't think `document` can be undefined. So is it okay to add/remove event listeners from `ownerDocument` instead of `document`?

![image](https://user-images.githubusercontent.com/45748283/191048941-13b64352-7e50-4377-9310-7ef1bd31b4ed.png)

![image](https://user-images.githubusercontent.com/45748283/191049400-5a048506-bedc-4e33-8a72-992844e00ff6.png)

Reply:
There can be multiple documents associated with a page, e.g. when a popup window is opened (using [`window.open()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/open)) and our components are used inside that popup. So we always want to add events to the correct document, and `ownerDocument` helps us do that.

And `document` can also be undefined when running this code _outside_ the browser (e.g. in [SSR](https://www.zachleat.com/web/ssr-overloaded/#component-framework-ssr) or in unit tests). This is why our [style guide](https://github.com/iTwin/iTwinUI-react/blob/main/STYLEGUIDE.md#use-getdocument-getwindow-instead-of-direct-access) recommends guarding document checks.

Those TypeScript types don't always account for code that runs outside the browser.